com.unity.multiplayer.mlapi/Runtime/NetworkVariable/Collections/NetworkList.cs

```diff
@@ -72,6 +72,8 @@ public NetworkList(IList<T> value)
             m_List = value;
         }
 
+        public string Name { get; set; }
```

Comment:
Just a thought, should this be `internal set;`? I don't know if users should be able to change the name of a network variable

Reply:
Unfortunately, since we need it in the interface, it can't be marked as internal.

Reply:
huh, why it can't be `public string Name { get; internal set; }`?
if there's an interface, it could also just be `string Name { get; }` to support the line above.

Reply:
I would usually fix this problem with appropriate member visibility as you're suggesting, but since is reflection code, we don't have that option.

Reply:
ah I see, we don't have our concrete type in the `NetworkBehaviour` on line `instance.Name = sortedFields[i].Name;`, we only have the interface so that we couldn't know about the internal setter.
since we're doing one-off runtime reflection in that method in NetworkBehaviour anyway, I'd suggest doing a little bit more reflection tricks to pull setter off of the property (not super fancy I know) but we might trade-off that unfanciness (is that a word?) against reducing the surface area for errors, especially on user-facing public APIs (and hopefully roslyn source generators might also help getting rid of runtime reflection codes in the future).

Reply:
I wouldn't dictate though, it's up to you. if it becomes really ugly with runtime reflection pulling off property setter method and calling it, you might not want to do that but just plugging my two cents here.

Reply:
The fix isn't too ugly after all, so I went ahead with it, since we're already using reflection in that section.

Also added tests to make sure we're not breaking anything.